flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SourceStreamTask.java

```diff
@@ -114,6 +114,7 @@ protected void cleanup() {
 	protected void performDefaultAction(ActionContext context) throws Exception {
 		// Against the usual contract of this method, this implementation is not step-wise but blocking instead for
 		// compatibility reasons with the current source interface (source functions run as a loop, not in steps).
+		sourceThread.setTaskDescription(getName());
```

Comment:
I wonder why not do it via `this.sourceThread = new LegacySourceFunctionThread(getName())` directly in master branch. Then 
```
LegacySourceFunctionThread(String name) {
       super(name);
}
```

Although `performDefaultAction` would be executed only once for source task, it seems better to do it in constructor. WDYT?

Reply:
Thanks for review @zhijiangW.

Afair, the story behind why it's done this way in the `master` branch, because the `LegacySourceFunctionThread` is instantiated in the constructor of `SourceStreamTask` and `StreamTask::getName()` method is not final.
To avoid a virtual method call in the constructor, @GJL has moved thread's name setup to the `performDefaultAction()` method just before starting the thread.

Reply:
Thanks for the explanation! 
Another option is via `env.getTaskInfo().getTaskNameWithSubtasks()` instead of `getName()` in constructor?
I could also accept the current way if you prefer since it was already accepted before in master. :)

Reply:
Using `env.getTaskInfo().getTaskNameWithSubtasks()` directly was also rejected in that story because of code duplication. 😀

I'd prefer to keep this change as similar to `master` branch as possible, to avoid backport conflicts for other PRs (if happens).